### PR TITLE
Testing controllers

### PR DESCRIPTION
**Pull request by Mesi21**
- Add the 'rails-controller-testing' gem
- Add tests for Users and Posts controllers
- Fix linter issues

### CODE REVIEW

File: spec/controllers/posts_spec.rb
Comment (pinkmoon25) [inline spec/controllers/posts_spec.rb:1]: - [x] Good work testing the controllers however, as mentioned in the requirements please make sure to test each aspect for each action for example test If the response status was correct, If a correct template was rendered, and If the response body includes the correct placeholder text for your `Posts Index action` and then all three aspect for `Posts Show action`, currently you have only one test case which checks the response status for` Posts Index action` and response includes correct placeholder for `Posts Show action`.
  - you are missing    
```
 it 'gets all posts from the user' do
        get '/users/1/posts'
        expect(response).to render_template(:index)
      end
      it 'gets all posts from the user' do
        get '/users/1/posts'
        expect(response.body).to include('List of all posts')
      end
```
from the Posts Index action and missing      
```
 it 'gets all posts from the user' do
        get '/users/1/posts/1'
        expect(response.body).to include('Info about a post')
      end
    it 'gets all posts from the user' do
      get '/users/1/posts'
      expect(response).to have_http_status(:success)
    end
```

from the Posts Shw action

File: spec/controllers/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      get '/users'
+      expect(response.status).to eq(200)
+      expect(response).to render_template('index')
+    end
+
+    it 'renders the show page with user details' do
+      get '/users/1'
+      expect(response.status).to eq(200)
+      expect(response).to render_template('show')
+    end
```
Comment (pinkmoon25) [inline spec/controllers/users_spec.rb]: - [ ] Again here we are missing the correct placeholder tests for both actions, let's try to fix this :+1:

**Review by pinkmoon25 — CHANGES_REQUESTED** [review]
Hi @Mesi21 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
Linter Checks :heavy_check_mark: 
Rspec-rails :heavy_check_mark: 
Requests specs :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omarsalem7 — APPROVED** [review]
Hi @Mesi21 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Add request changes by the previous reviewer
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**